Intersect.Client/Interface/Game/Map/MinimapWindow.cs

```diff
@@ -0,0 +1,571 @@
+                    }
+                }
+
+                _redrawMaps = true;
+            }
+
+            var newLocations = GenerateEntityInfo(allEntities, player);
+            if (!newLocations.Equals(_entityInfoCache))
+            {
+                _entityInfoCache = newLocations;
+                _redrawEntities = true;
+            }
+
+            // Update our minimap display area
+            var centerX = (_renderTexture.Width / 3) + (player.X * _minimapTileSize.X);
+            var centerY = (_renderTexture.Height / 3) + (player.Y * _minimapTileSize.Y);
+            var displayWidth = (int)(_renderTexture.Width * (_zoomLevel / 100f));
+            var displayHeight = (int)(_renderTexture.Height * (_zoomLevel / 100f));
+
+            var x = centerX - (displayWidth / 2);
+            if (x + displayWidth > _renderTexture.Width)
+            {
+                x = _renderTexture.Width - displayWidth;
+            }
+
+            if (x < 0)
+            {
+                x = 0;
+            }
+
+            var y = centerY - (displayHeight / 2);
+            if (y + displayHeight > _renderTexture.Height)
+            {
+                y = _renderTexture.Height - displayHeight;
+            }
+
+            if (y < 0)
+            {
+                y = 0;
+            }
+
+            _minimap.SetTextureRect(x, y, displayWidth, displayHeight);
+        }
+
+        private void DrawMinimap()
+        {
+            if (!_redrawEntities && !_redrawMaps)
+            {
+                return;
+            }
+            
+            _renderTexture.Clear(Color.Transparent);
+
+            foreach (var pos in _mapGrid.Keys)
+            {
+                if (_redrawMaps)
+                {
+                    GenerateMinimapCacheFor(pos);
+                }
+
+                if (_redrawEntities)
+                {
+                    GenerateEntityCacheFor(pos);
+                }
+
+                DrawMinimapCacheToTexture(pos);
+            }
+
+            if (_redrawMaps)
+            {
+                _redrawMaps = false;
+            }
+
+            if (_redrawEntities)
+            {
+                _redrawEntities = false;
+            }
+        }
+
+        private void GenerateMinimapCacheFor(MapPosition position)
+        {
+            _minimapCache.TryAdd(position, null);
+            _minimapCache[position] ??= GenerateMapRenderTexture();
+            _minimapCache[position].Clear(Color.Transparent);
+
+            if (_mapGrid[position] == null)
+            {
+                return;
+            }
+
+            foreach (var layer in Options.Instance.MinimapOpts.RenderLayers)
+            {
+                for (var x = 0; x < Options.Instance.MapOpts.MapWidth; x++)
+                {
+                    for (var y = 0; y < Options.Instance.MapOpts.MapHeight; y++)
+                    {
+                        var curTile = _mapGrid[position].Layers[layer][x, y];
```

Comment:
`_mapGrid[position]` is being looked up 4160 times on the default map width (32)/map height (26)/render layers (5) settings. It needs to be looked up 1 time. :| Cache this locally before checking it for null on L204.

Reply:
done 👍🏽  !